src/man/macdaily.1
.\" Man page generated from reStructuredText.
.
.TH MACDAILY 8 "March 31, 2019" "v2019.3.31.post1" ""
.SH NAME
macdaily \- macOS Automated Package Manager
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.SH SYNOPSIS
.sp
macdaily [\fIoptions\fP] <\fIcommand\fP> ...
.SH DESCRIPTION
.sp
\fBMacDaily\fP is an all\-in\-one collection of console utility written in Python
with support of \fI\%PTY\fP\&.
Originally works as an automated housekeeper for Mac to update all packages
outdated, \fBMacDaily\fP is now fully functioned and end\-user oriented.
.sp
\fBMacDaily\fP can manage packages of various distributions of
\fI\%Atom\fP, \fI\%RubyGems\fP,
\fI\%Homebrew\fP, \fI\%Python\fP,
\fI\%Node.js\fP, and even macOS software updates (c.f.
\fBsoftwareupdate(8)\fP). Without being aware of everything about your Mac, one
can easily work around and manage packages out of no pain using \fBMacDaily\fP\&.
.SH EXAMPLES
.INDENT 0.0
.TP
.B NOTE
all acronyms and aliases are left out for a quick and
clear view of \fBMacDaily\fP
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to use \fBMacDaily\fP?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
# call from PATH
$ macdaily <command> [option ...]
# or call as Python module
$ python \-m macdaily <command> [option ...]
# or call a certain command
$ md\-${command} [option ...]
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to set up my disks and daemons interactively?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily config \-\-interactive
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to relaunch daemons after I manually modified \fI~/.dailyrc\fP?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily launch daemons
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to archive all ancient logs without running any commands?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily archive \-\-all
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to update all outdated packages?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily update \-\-all
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to update a certain package (eg: \fIhello\fP from Homebrew)?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily update brew \-\-packages=hello
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to update without a certain package (eg: update all packages
except Python package \fIptyng\fP)?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily update \-\-all \-\-pip=\(aq!ptyng\(aq
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to uninstall a certain package along with its dependencies (eg:
\fIpytest\fP from brewed CPython version 3.6)?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily uninstall pip \e
      \-\-brew \-\-cpython \-\-python=3\&.6 \-\-packages=pytest
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to reinstall all packages but do not cleanup caches?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily reinstall \-\-all \-\-no\-cleanup
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to postinstall packages whose name ranges between \fIstart\fP and
\fIstop\fP alphabetically?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily postinstall \-\-all \-\-start=start \-\-end=stop
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to show dependency of a certain package as a tree (eg: \fIgnupg\fP
from Homebrew) ?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily dependency brew  \-\-tree \-\-packages=gnupg
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to log all applications on my Mac, a.k.a. \fI*.app\fP files?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily logging dotapp
.ft P
.fi
.UNINDENT
.UNINDENT
.INDENT 0.0
.IP \(bu 2
How to dump a \fIMacfile\fP to keep track of all packages?
.UNINDENT
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ macdaily bundle dump
.ft P
.fi
.UNINDENT
.UNINDENT
.SH OPTIONS
.SS optional arguments
.INDENT 0.0
.TP
.B \-h\fP,\fB  \-\-help
show this help message and exit
.TP
.B \-V\fP,\fB  \-\-version
show program\(aqs version number and exit
.TP
.B \-E\fP,\fB  \-\-environ
show all available environment variables
.UNINDENT
.SS command selection
.sp
MacDaily provides a friendly CLI workflow for the administrator of macOS
to manipulate packages, see \fBmacdaily commands\fP for more information
.INDENT 0.0
.TP
.B archive
archive ancient runtime logs
.TP
.B bundle
bundler for all packages on your Mac
.TP
.B cleanup
remove outdated downloads, caches, etc.
.TP
.B commands
show available commands and corresponding subsidiaries
.TP
.B config
get and set MacDaily runtime options
.TP
.B dependency
show dependencies for packages
.TP
.B help
show man pages for \fIcommand\fP
.TP
.B install
install packages
.TP
.B launch
launch daemon services and helper programs
.TP
.B logging
record packages on your Mac
.TP
.B postinstall
run the post\-install steps for Homebrew \fIformula\fP
.TP
.B reinstall
reinstall existing packages
.TP
.B uninstall
recursively uninstall packages
.TP
.B update
update packages
.UNINDENT
.SH ENVIRONMENT
.sp
MacDaily currently supports two environment variables.
For boolean values, MacDaily currently uses the same
mapping as \fBconfigparser.ConfigParser.getboolean\fP
function in Python.
.TS
center;
|l|l|l|l|.
_
T{
Value
T}	T{
Boolean
T}	T{
Value
T}	T{
Boolean
T}
_
T{
\fB1\fP
T}	T{
\fBTrue\fP
T}	T{
\fB0\fP
T}	T{
\fBFalse\fP
T}
_
T{
\fByes\fP
T}	T{
\fBTrue\fP
T}	T{
\fBno\fP
T}	T{
\fBFalse\fP
T}
_
T{
\fBtrue\fP
T}	T{
\fBTrue\fP
T}	T{
\fBfalse\fP
T}	T{
\fBFalse\fP
T}
_
T{
\fBon\fP
T}	T{
\fBTrue\fP
T}	T{
\fBoff\fP
T}	T{
\fBFalse\fP
T}
_
.TE
.INDENT 0.0
.TP
.B NOTE
environment variables must have a value set  to  be  detected.
For  example, \fBexport MACDAILY_DEVMODE=1\fP rather than just
\fBexport MACDAILY_DEVMODE\fP\&.
.TP
.B SUDO_PASSWORD
password of your current account (for \fBsudo(8)\fP command)
.TP
.B NULL_PASSWORD
implies \fBSUDO_PASSWORD=\(aq\(aq\fP and \fBMACDAILY_NO_CHECK=true\fP
.sp
\fIdefault\fP: \fBfalse\fP
.TP
.B MACDAILY_NO_CHECK
do not validate your password (for \fBsudo(8)\fP command)
.sp
\fIdefault\fP: \fBfalse\fP
.TP
.B MACDAILY_NO_CONFIG
do not load configuration from \fB~/.dailyrc\fP
.sp
\fIdefault\fP: \fBfalse\fP
.TP
.B MACDAILY_LOGDIR
path where logs will be stored
.sp
\fIdefault\fP: \fB~/Library/Logs/MacDaily\fP
.TP
.B MACDAILY_DSKDIR
path where your hard disk lies
.TP
.B MACDAILY_ARCDIR
path where ancient logs archive; log archives will be named
as \fBarchive.zip\fP, while other archives (e.g. Homebrew
caches) will remain in directories as where they were from.
.sp
\fIdefault\fP: \fB${MACDAILY_DSKDIR}/Developers\fP
.TP
.B MACDAILY_LIMIT
timeout limit for shell commands in seconds
.sp
\fIdefault\fP: \fB1,000\fP
.TP
.B MACDAILY_RETRY
retry timeout for input prompts in seconds
.sp
\fIdefault\fP: \fB60\fP
.TP
.B MACDAILY_CLEANUP
run cleanup process after any operation (MacDaily command)
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_APM
disable operations (MacDaily commands) on \fBAtom plug\-ins\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_APP
disable operations (MacDaily commands) on \fBmacOS applications\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_BREW
disable operations (MacDaily commands) on \fBHomebrew Formulae\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_CASK
disable operations (MacDaily commands) on \fBHomebrew Casks\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_GEM
disable operations (MacDaily commands) on \fBRuby gems\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_MAS
disable operations (MacDaily commands) on \fBMac App Store applications\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_NPM
disable operations (MacDaily commands) on \fBNode.js modules\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_PIP
disable operations (MacDaily commands) on \fBPython packages\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_system
disable operations (MacDaily commands) on \fBmacOS software\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_TAP
disable operations (MacDaily commands) on \fBHomebrew Taps\fP
.sp
\fIdefault\fP: \fBtrue\fP
.TP
.B MACDAILY_DEVMODE
enabled development mode (\fIonly for debugging\fP)
.sp
\fIdefault\fP: \fBfalse\fP
.UNINDENT
.SH SEE ALSO
.INDENT 0.0
.IP \(bu 2
MacDaily documentation: \fIhttps://github.com/JarryShaw/MacDaily#generals\fP
.IP \(bu 2
MacDaily Log Archive Utility
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-archive\fP
.UNINDENT
.IP \(bu 2
macOS Package Cache Cleanup
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-cleanup\fP
.IP \(bu 2
\fBmacdaily\-cleanup\-brew\fP
.IP \(bu 2
\fBmacdaily\-cleanup\-cask\fP
.IP \(bu 2
\fBmacdaily\-cleanup\-npm\fP
.IP \(bu 2
\fBmacdaily\-cleanup\-pip\fP
.UNINDENT
.IP \(bu 2
MacDaily Runtime Configuration Helper
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-config\fP
.UNINDENT
.IP \(bu 2
macOS Package Dependency Query
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-dependency\fP
.IP \(bu 2
\fBmacdaily\-dependency\-brew\fP
.IP \(bu 2
\fBmacdaily\-dependency\-pip\fP
.UNINDENT
.IP \(bu 2
MacDaily Usage Information Manual
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-help\fP
.UNINDENT
.IP \(bu 2
macOS Package Automated Installer
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-install\fP
.IP \(bu 2
\fBmacdaily\-install\-apm\fP
.IP \(bu 2
\fBmacdaily\-install\-brew\fP
.IP \(bu 2
\fBmacdaily\-install\-cask\fP
.IP \(bu 2
\fBmacdaily\-install\-gem\fP
.IP \(bu 2
\fBmacdaily\-install\-mas\fP
.IP \(bu 2
\fBmacdaily\-install\-npm\fP
.IP \(bu 2
\fBmacdaily\-install\-pip\fP
.IP \(bu 2
\fBmacdaily\-install\-system\fP
.UNINDENT
.IP \(bu 2
MacDaily Dependency Launch Helper
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-launch\fP
.UNINDENT
.IP \(bu 2
macOS Package Logging Automator
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-logging\fP
.IP \(bu 2
\fBmacdaily\-logging\-apm\fP
.IP \(bu 2
\fBmacdaily\-logging\-app\fP
.IP \(bu 2
\fBmacdaily\-logging\-brew\fP
.IP \(bu 2
\fBmacdaily\-logging\-cask\fP
.IP \(bu 2
\fBmacdaily\-logging\-gem\fP
.IP \(bu 2
\fBmacdaily\-logging\-mas\fP
.IP \(bu 2
\fBmacdaily\-logging\-npm\fP
.IP \(bu 2
\fBmacdaily\-logging\-pip\fP
.IP \(bu 2
\fBmacdaily\-logging\-tap\fP
.UNINDENT
.IP \(bu 2
Homebrew Cask Postinstall Automator
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-postinstall\fP
.UNINDENT
.IP \(bu 2
Automated macOS Package Reinstaller
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-reinstall\fP
.IP \(bu 2
\fBmacdaily\-reinstall\-brew\fP
.IP \(bu 2
\fBmacdaily\-reinstall\-cask\fP
.UNINDENT
.IP \(bu 2
Automated macOS Package Uninstaller
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-uninstall\fP
.IP \(bu 2
\fBmacdaily\-uninstall\-brew\fP
.IP \(bu 2
\fBmacdaily\-uninstall\-cask\fP
.IP \(bu 2
\fBmacdaily\-uninstall\-pip\fP
.UNINDENT
.IP \(bu 2
macOS Package Update Automator
.INDENT 2.0
.IP \(bu 2
\fBmacdaily\-update\fP
.IP \(bu 2
\fBmacdaily\-update\-apm\fP
.IP \(bu 2
\fBmacdaily\-update\-brew\fP
.IP \(bu 2
\fBmacdaily\-update\-cask\fP
.IP \(bu 2
\fBmacdaily\-update\-gem\fP
.IP \(bu 2
\fBmacdaily\-update\-mas\fP
.IP \(bu 2
\fBmacdaily\-update\-npm\fP
.IP \(bu 2
\fBmacdaily\-update\-pip\fP
.IP \(bu 2
\fBmacdaily\-update\-system\fP
.UNINDENT
.UNINDENT
.SH BUGS
.sp
If any bugs, please file issues on GitHub:
.INDENT 0.0
.TP
.B JarryShaw/MacDaily
\fI\%https://github.com/JarryShaw/MacDaily/issues\fP
.UNINDENT
.sp
Contribution is welcome.
.SH AUTHOR
Jarry Shaw, a newbie programmer, is the author, owner and maintainer
of MacDaily. Please contact me at jarryshaw@icloud.com.
.SH COPYRIGHT
MacDaily is licensed under the Apple Open Source License.
.\" Generated by docutils manpage writer.
.


CHANGELOG.md
# Change Log

## [v2019.3.31.post1](https://github.com/JarryShaw/MacDaily/tree/v2019.3.31.post1) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.31.post1.-devel...v2019.3.31.post1)

## [v2019.3.31.post1.-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.31.post1.-devel) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.30.e89c90-devel...v2019.3.31.post1.-devel)

**Merged pull requests:**

-  New distribution \[2019.03.31.post1\] [\#55](https://github.com/JarryShaw/MacDaily/pull/55) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.30.e89c90-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.30.e89c90-devel) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.03.30...v2019.3.30.e89c90-devel)

## [v2019.03.30](https://github.com/JarryShaw/MacDaily/tree/v2019.03.30) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.30...v2019.03.30)

## [v2019.3.30](https://github.com/JarryShaw/MacDaily/tree/v2019.3.30) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.03.31...v2019.3.30)

## [v2019.03.31](https://github.com/JarryShaw/MacDaily/tree/v2019.03.31) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.31.1b8496-devel...v2019.03.31)

**Merged pull requests:**

- New distribution \[2019.03.31\] [\#54](https://github.com/JarryShaw/MacDaily/pull/54) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.31.1b8496-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.31.1b8496-devel) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.31...v2019.3.31.1b8496-devel)

## [v2019.3.31](https://github.com/JarryShaw/MacDaily/tree/v2019.3.31) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.31.e89c90-devel...v2019.3.31)

**Merged pull requests:**

- New distribution \[2019.03.30\] [\#53](https://github.com/JarryShaw/MacDaily/pull/53) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.31.e89c90-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.31.e89c90-devel) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.31.-devel...v2019.3.31.e89c90-devel)

## [v2019.3.31.-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.31.-devel) (2019-03-30)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.28.post1...v2019.3.31.-devel)

## [v2019.3.28.post1](https://github.com/JarryShaw/MacDaily/tree/v2019.3.28.post1) (2019-03-28)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.28.post1.1baf18-devel...v2019.3.28.post1)

**Merged pull requests:**

- Bugfix [\#52](https://github.com/JarryShaw/MacDaily/pull/52) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.28.post1.1baf18-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.28.post1.1baf18-devel) (2019-03-28)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.28...v2019.3.28.post1.1baf18-devel)

## [v2019.3.28](https://github.com/JarryShaw/MacDaily/tree/v2019.3.28) (2019-03-28)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.28.edfd7e-devel...v2019.3.28)

**Merged pull requests:**

- New distribution \[2019.03.28\] [\#51](https://github.com/JarryShaw/MacDaily/pull/51) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.28.edfd7e-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.28.edfd7e-devel) (2019-03-28)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.12...v2019.3.28.edfd7e-devel)

## [v2019.3.12](https://github.com/JarryShaw/MacDaily/tree/v2019.3.12) (2019-03-12)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.12.899c64-devel...v2019.3.12)

**Merged pull requests:**

- New distribution \[2019.3.13\] [\#50](https://github.com/JarryShaw/MacDaily/pull/50) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.12.899c64-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.12.899c64-devel) (2019-03-12)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.10...v2019.3.12.899c64-devel)

## [v2019.3.10](https://github.com/JarryShaw/MacDaily/tree/v2019.3.10) (2019-03-10)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.10.0e7ce2-devel...v2019.3.10)

**Merged pull requests:**

- New distribution \[2019.3.10\] [\#49](https://github.com/JarryShaw/MacDaily/pull/49) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.10.0e7ce2-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.10.0e7ce2-devel) (2019-03-10)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.9...v2019.3.10.0e7ce2-devel)

## [v2019.3.9](https://github.com/JarryShaw/MacDaily/tree/v2019.3.9) (2019-03-09)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.9.76cbfe-devel...v2019.3.9)

**Merged pull requests:**

- New distribution \[2019.03.09\] [\#48](https://github.com/JarryShaw/MacDaily/pull/48) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.9.76cbfe-devel](https://github.com/JarryShaw/MacDaily/tree/v2019.3.9.76cbfe-devel) (2019-03-09)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.8.post1...v2019.3.9.76cbfe-devel)

**Merged pull requests:**

- New distribution \[2019.03.08.post2\] [\#47](https://github.com/JarryShaw/MacDaily/pull/47) ([JarryShaw](https://github.com/JarryShaw))
- New distribution \[2019.03.08.post1\] [\#46](https://github.com/JarryShaw/MacDaily/pull/46) ([JarryShaw](https://github.com/JarryShaw))

## [v2019.3.8.post1](https://github.com/JarryShaw/MacDaily/tree/v2019.3.8.post1) (2019-03-08)
[Full Changelog](https://github.com/JarryShaw/MacDaily/compare/v2019.3.8.post1.ca00e0-devel...v2019.3.8.post1)

**Merged pull requests:**

- New distribution \[2019.03.08.post1\] [\#45](https://github.com/JarryShaw/MacDaily/pull/45) ([JarryShaw](https://github.com/JarryShaw))


Formula/macdaily.rb
class Macdaily < Formula
  include Language::Python::Virtualenv

  desc "macOS Automated Package Manager"
  homepage "https://github.com/JarryShaw/MacDaily#macdaily"
  url "https://files.pythonhosted.org/packages/53/41/7ef5c147f99f0f7acee244e2d44922020c24e90e8613c85ddcb231e6e65e/macdaily-2018.11.25.post1.tar.gz"
  sha256 "2eefe8fd381f9ec8559293ae16dc4fa782a56c607c46e1807fdce384d24fc862"
  head "https://github.com/JarryShaw/MacDaily.git", :branch => "release"

  bottle do
    cellar :any_skip_relocation
    sha256 "2eefe8fd381f9ec8559293ae16dc4fa782a56c607c46e1807fdce384d24fc862" => :mojave
  end

  # devel do
  #   url "https://github.com/JarryShaw/MacDaily.git"
  #   sha256 "85cc828a96735bdafcf29eb6291ca91bac846579bcef7308536e0c875d6c81d7"
  # end

  depends_on "python"

  depends_on "expect" => :recommended

  resource "configupdater" do
    url "https://files.pythonhosted.org/packages/54/b8/1aa82f89b77045c897c35d3f2f28c7559282fd422018c0377efc362f56d1/ConfigUpdater-0.3.2.tar.gz"
    sha256 "f8e832a1a40faca6aa9abdc3069cc0fb9021580dcef96cbe1aad055bc08ab865"
  end

  resource "dictdumper" do
    url "https://files.pythonhosted.org/packages/e7/f7/fa8b1c68cb02270e58852c178d4d4eea099e2806e0da9ca7fcc7920b4463/dictdumper-0.6.5.post2.tar.gz"
    sha256 "96d4d0910662d8e21e2c1900bb5438dbb2562e680bda34be2d9cf1434e8b0416"
  end

  resource "psutil" do
    url "https://files.pythonhosted.org/packages/e3/58/0eae6e4466e5abf779d7e2b71fac7fba5f59e00ea36ddb3ed690419ccb0f/psutil-5.4.8.tar.gz"
    sha256 "6e265c8f3da00b015d24b842bfeb111f856b13d24f2c57036582568dc650d6c3"
  end

  resource "ptyng" do
    url "https://files.pythonhosted.org/packages/02/4b/2d6c1543657cddf4a061aa9fb86e20b354706e37868a76d9691740d1fd16/ptyng-0.2.1.post1.tar.gz"
    sha256 "a199c0a60eaf3d9e10aad97d858f0feb032e87ccdefe80592146cc95b65362fd"
  end

  def install
    virtualenv_install_with_resources
    man1.install "macdaily/man/macdaily.1"
  end

  def post_install
    system bin/"macdaily" "launch" "--all"
  end

  def caveats
    text = <<~EOS
      MacDaily has been installed as
        #{HOMEBREW_PREFIX}/bin/macdaily

      Alias executables `md-update`, `md-uninstall`, etc. equal to
      `macdaily update`, `macdaily uninstall`, etc., respectively,
      have been also installed into #{HOMEBREW_PREFIX}/bin/

      Configuration file locates at ~/.dailyrc, please directly run
      `macdaily config --interactive` command to set up your runtime
      specifications.

      For more information, check out `macdaily help` command. Online
      documentations available at GitHub repository.

      See: https://github.com/JarryShaw/MacDaily#generals
    EOS
    text
  end

  test do
    system bin/"macdaily", "--help"
  end
end
